src/middlewared/middlewared/plugins/account.py

```diff
@@ -1445,6 +1448,132 @@ def update_sshpubkey(self, homedir, user, group):
             os.fchown(f.fileno(), user['uid'], gid)
             f.write(f'{pubkey}\n')
 
+    @no_authz_required
+    @accepts(Dict(
+        'password_reset_data',
+        Str('username', required=True),
+        Password('old_password', default=''),
+        Password('new_password', required=True),
+        Dict(
+            'options',
+            Bool('skip_password_check', default=False)
```

Comment:
We can have a simpler API interface: make `old_password` not required on schema level, but check that it is set if we are not a full admin. Then this flag won't be necessary.